### DISCUSSION TITLE
only image is not rendering in nextjs  rect circle  are working this my sample code

### DISCUSSION THREAD

**Discussion by randomtesting6061** [fabricjs/fabric.js#10057] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/25b1df6c-aa3b-4d4d-934a-b0e2acf1c91b)

"use client"

import React, { useRef, useEffect } from 'react';
// import { Canvas, Rect,Image } from 'fabric'; 
import * as fabric from 'fabric'
/// components/FabricCanvas.js

const FabricCanvas = () => {
  const canvasRef = useRef(null);

  useEffect(() => {
    const canvas = new fabric.Canvas(canvasRef.current);

    // Create a white background rectangle
    const backgroundRect = new fabric.Rect({
      left: 0,
      top: 0,
      fill: 'white', // Set the background color to white
      width: canvas.width,
      height: canvas.height,
      selectable: false, // Make sure it's not selectable
    });
    const backgroundRect2 = new fabric.Rect({
      left: 300,
      top: 0,
      fill: 'red', // Set the background color to white
      width: 50,
      height:50,
      selectable: false, // Make sure it's not selectable
    });

    // Add the background rectangle to the canvas
    canvas.add(backgroundRect);
    console.log('Background added:', backgroundRect);

    // Load the image
    fabric.FabricImage.fromURL(
      'https://via.placeholder.com/500',
      (img) => {
        console.log('Loaded image object:');
        img.left = 0;
        img.top = 0;
        img.visible = true;
        canvas.add(img);
        canvas.centerObject(img);
        canvas.renderAll();
      },
      { crossOrigin: 'anonymous' }
    );
    
    canvas.add(backgroundRect2);
    canvas.renderAll()
    return () => {
      canvas.dispose();
    };
  }, []);

  return (
    <div style={{ display: 'flex', justifyContent: 'center', alignItems: 'center', height: '100vh' }}>
      <canvas ref={canvasRef} width={500} height={500} style={{ border: '1px solid #ccc' }}></canvas>
    </div>
  );
};

export default FabricCanvas;

**Comment by zhe-he** [comment, marked as answer]
The image does not support cross-origin requests; you need to set the image to allow cross-origin access.

<img width="509" alt="image" src="https://github.com/user-attachments/assets/b259ca9b-de94-4068-93b4-d6698245519b">

[demo](https://codepen.io/zhe-he-the-vuer/pen/oNrGVaO?editors=1111)

**Reply by randomtesting6061** [reply]
now it working fine i degraded the version i think it is problem with version 6 thankyou

**Reply by asturur** [reply]
your code can't work with v6 because is using callbacks.
